### PR TITLE
UIFC-366: Remove dependencies of k-int and stripes-erm-components

### PR DESCRIPTION
**Pull request by elsenhans**
https://folio-org.atlassian.net/browse/UIFC-366

### CODE REVIEW

File: src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js:1]: The only Form we do have is in Filters and the only use of EditCard is here. So there is no other place using it right now.

File: src/components/Filters/FilterFile/UploadFile/EditCard/editcard-util.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/editcard-util.js:1]: Thats right, but those fields are used for the EditCard.

File: src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js:1]: You would prefer to rename it to `fields-util.js`?

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        <Form
+          onSubmit={jest.fn}
+          mutators={{
+            selectedContact,
+            ...arrayMutators,
+          }}
+          render={() => withIntlConfiguration(
+            <DocumentsFieldArray fields={{ name: 'filterFiles' }}>
+              <Field component={TextField} />
+              <Field component={TextField} />
```
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js]: Removed

File: src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.test.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.test.js:1]: Added a test for checking the children.

File: src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/EditCard.js:1]: Moved EditCard to /DisplayUtils/

File: src/components/Filters/FilterFile/UploadFile/EditCard/editcard-util.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/EditCard/editcard-util.js:1]: renamed to _fieldArray-util.js_

File: src/components/DisplayUtils/EditCard/fieldArray-util.js
Comment (alb3rtino) [inline src/components/DisplayUtils/EditCard/fieldArray-util.js:1]: The `onUpdateField` method is not used anywhere and can be removed.
`field?.id` in the `onDeleteField` method is always undefined in your use case.

This leaves us with a module that only defines new methods for `fields.push` and `fields.remove`.

Please delete this module and use `fields.push` and `fields.remove` directly in `<DocumentsFieldArray>`.

**Review by alb3rtino — CHANGES_REQUESTED** [review]

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js [hunk 2]
```diff
@@ -68,8 +72,8 @@ const DocumentsFieldArray = ({
           'data-test-delete-filter-file-button': true
         }}
         header={<FormattedMessage id="ui-finc-select.filter.file.label" values={{ number: i + 1 }} />}
-        key={i}
-        onDelete={() => onDeleteField(i, doc)}
+        key={fields.value[i].fileId}
```
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js:71]: Do you have a suggestion for a solution?

File: src/components/DisplayUtils/EditCard/fieldArray-util.js
Comment (elsenhans) [reply src/components/DisplayUtils/EditCard/fieldArray-util.js:1]: removed

File: src/components/DisplayUtils/EditCard/EditCard.test.js
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/DisplayUtils/EditCard/EditCard.test.js:1]: checkig if onDelete has been called

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js (hunk 2 above)
Comment (alb3rtino) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js:71]: no. best leave it as it was and deal with it later.

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const selectedContact = jest.fn();
+
+const renderDocumentsFieldArray = (stripes) => {
+  return render(withIntlConfiguration(
+    <StripesContext.Provider value={stripes}>
+      <MemoryRouter>
+        <Form
+          onSubmit={jest.fn}
+          mutators={{
+            selectedContact,
```
Comment (alb3rtino) [inline src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js:21]: Cant see that this mutator is required for anything?

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js (hunk 2 above)
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.js]: reset the changes

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            ...arrayMutators,
+          }}
+          render={() => withIntlConfiguration(
```
Comment (alb3rtino) [inline src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js:24]: `withIntlConfiguration()` is used twice.

**Review by alb3rtino — CHANGES_REQUESTED** [review]

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import withIntlConfiguration from '../../../../../test/jest/helpers/withIntlConfiguration';
+
+const selectedContact = jest.fn();
+
+const renderDocumentsFieldArray = (stripes) => {
+  return render(withIntlConfiguration(
+    <StripesContext.Provider value={stripes}>
+      <MemoryRouter>
+        <Form
+          onSubmit={jest.fn}
```
Comment (alb3rtino) [inline src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js:19]: should be `jest.fn()`

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js (hunk 3 above)
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js]: selectedContact removed

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js (hunk 4 above)
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js]: removed one withIntlConfiguration

File: src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js (hunk 5 above)
Comment (elsenhans) [reply src/components/Filters/FilterFile/UploadFile/DocumentsFieldArray.test.js]: added parentheses

**Review by alb3rtino — APPROVED** [review]

**Pull request merged**